labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -29,39 +29,59 @@ object RecursionPatternMatchingExercise {
    * checkValuesIncrease(Seq(1,2,2)) == false
    */
   def checkValuesIncrease(seq: Seq[Int]): Boolean = {
-    error("fix me")
+    if (seq.length == 0) return false
+    if (seq.length == 1) return true
+    else
+      (seq(0) < seq(1)) match {
+      case true => return checkValuesIncrease(seq.tail)
+      case false => return false
+    }
   }
   
   /**
    * Group Consecutive values
    * List(1,1,2,3,1,1) -> List(1,1), List(2), List(3), List(1,1)
    */
-  def groupConsecutive[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+  def groupConsecutive[T](in: List[T]): List[List[T]] = in.length match{
+    case 0 => List()
+    case 1 => List(in)
+    case _ => subGroup(in) :: groupConsecutive(in.drop(subGroup(in).length))
+  }
+  
+  def subGroup[T](in: List[T]): List[T] = in.length match {
+    case 0 => List()
+    case 1 => in
+    case _ => in takeWhile {in.head ==}
   }
 
   /**
    * Group Equal values
    * List(1,1,2,3,1,1) -> List(1,1,1,1), List(2), List(3)
    */
-  def groupEquals[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+  def groupEquals[T](in: List[T]): List[List[T]] = in.length match{
+    case 0 => List()
+    case 1 => List(in)
+    case _ => List(in.filter((item: T) => item == in.head)) ::: groupEquals(in.filter((item: T) => item != in.head))
   }
 
   /**
    * Compress values
    * List(1,1,2,3,1,1) -> List(1,2,3)
    */
-  def compress[T](in: List[T]): List[T] = {
-    error("fix me")
+  def compress[T](in: List[T]): List[T] = in.length match{
+    case 0 => List()
+    case 1 => in
+    case _ => List(in.head) ::: compress(in.filter((item: T) => item != in.head))
   }
   
   /**
    * Define the amount of all equal members
    * List(1,1,2,3,1,1) -> List((4,1),(1,2),(1,3))
    */
-  def amountEqualMembers[T](in: List[T]): List[(Int, T)] = {
-    error("fix me")
+  def amountEqualMembers[T](in: List[T]): List[(Int, T)] = in.length match{
+    case 0 => List()
+    case 1 => List((1, in.head))
+    case _ => List(((in.filter((item: T) => item == in.head)).length, in.head)) ::: amountEqualMembers(in.filter((item: T) => item != in.head))
```

Comment:
Complex, but seemingly correct.

`Partition` may have been useful here. It separates list elements into two lists based on whether they meet some criteria